benches/bench.rs

```diff
@@ -134,40 +135,37 @@ impl RequestBencher for AsyncBencher {
 
 fn run_round_trip(rt: &TokioRuntime, crit: &mut Criterion, client: Arc<impl Client>, name: &str, request: RequestType) {
 	crit.bench_function(&request.group_name(name), |b| {
-		b.iter(|| {
-			rt.block_on(async {
-				black_box(client.request::<String>(request.method_name(), JsonRpcParams::NoParams).await.unwrap());
-			})
+		b.to_async(rt).iter(|| async {
+			black_box(client.request::<String>(request.method_name(), JsonRpcParams::NoParams).await.unwrap());
 		})
 	});
 }
 
 fn run_sub_round_trip(rt: &TokioRuntime, crit: &mut Criterion, client: Arc<impl SubscriptionClient>, name: &str) {
 	let mut group = crit.benchmark_group(name);
 	group.bench_function("subscribe", |b| {
-		b.iter_with_large_drop(|| {
-			rt.block_on(async {
-				black_box(
-					client
-						.subscribe::<String>(SUB_METHOD_NAME, JsonRpcParams::NoParams, UNSUB_METHOD_NAME)
-						.await
-						.unwrap(),
-				);
-			})
+		b.to_async(rt).iter_with_large_drop(|| async {
+			black_box(
+				client.subscribe::<String>(SUB_METHOD_NAME, JsonRpcParams::NoParams, UNSUB_METHOD_NAME).await.unwrap(),
+			);
 		})
 	});
 	group.bench_function("subscribe_response", |b| {
-		b.iter_with_setup(
+		b.to_async(rt).iter_with_setup(
 			|| {
-				rt.block_on(async {
-					client
-						.subscribe::<String>(SUB_METHOD_NAME, JsonRpcParams::NoParams, UNSUB_METHOD_NAME)
-						.await
-						.unwrap()
+				// We have to use `block_in_place` here since `b.to_async(rt)` automatically enters the
+				// runtime context and simply calling `block_on` here will cause the code to panic.
+				tokio::task::block_in_place(|| {
```

Comment:
hmm, I don't understand this.

Why doesn't this work:

```rust
async {
   client.subscribe::<String>(SUB_METHOD_NAME, JsonRpcParams::NoParams, UNSUB_METHOD_NAME)
      .await
	  .unwrap()
}
```

Reply:
Because for some reason in `iter_with_setup` the first closure [must be sync](https://github.com/bheisler/criterion.rs/blob/master/src/bencher.rs#L517), so the suggested way will just result in a future being returned (but not polled).

Reply:
ok, that makes sense.